llama/llama.go

```diff
@@ -680,3 +683,27 @@ func (s *SamplingContext) Sample(llamaContext *Context, idx int) int {
 func (s *SamplingContext) Accept(id int, applyGrammar bool) {
 	C.gpt_sampler_caccept(s.c, C.llama_token(id), C.bool(applyGrammar))
 }
+
+// KvCacheTypeFromStr converts a string cache type to the corresponding GGML type value
+func KvCacheTypeFromStr(s string) C.enum_ggml_type {
+	switch s {
+	case "f32":
+		return C.GGML_TYPE_F32
+	case "f16":
+		return C.GGML_TYPE_F16
+	case "q8_0":
+		return C.GGML_TYPE_Q8_0
+	case "q4_0":
+		return C.GGML_TYPE_Q4_0
+	case "q4_1":
+		return C.GGML_TYPE_Q4_1
+	case "iq4_nl":
+		return C.GGML_TYPE_IQ4_NL
+	case "q5_0":
+		return C.GGML_TYPE_Q5_0
+	case "q5_1":
+		return C.GGML_TYPE_Q5_1
+	default:
+		panic("Unsupported cache type: " + s)
```

Comment:
Returning an error would be preferred vs `panic`ing here

Reply:
Yeah good call, brain fart moment 😅